### ISSUE TITLE
Add fp16 PyTorch models

### ISSUE THREAD

**Issue by kkontny** [AmpereComputingAI/ampere_model_library#152]
In Pytorch you can convert model to fp16 with `module.half()` call. I think it should be called before converting to TorchScript. See docs https://pytorch.org/docs/stable/generated/torch.nn.Module.html. I think it should be quite simple implementation.

**Comment by jan-grzybek-ampere** [comment]
@dkupnicki please take a look

**Comment by dkupnicki** [comment]
I added `self.__model.half()` before https://github.com/AmpereComputingAI/ampere_model_library/blob/209549f127ef3ed642c2161598e54c46bb388fab/utils/pytorch.py#L46

and tested a few models. All of them stopped working, but they threw slightly different errors:
- resnet_50_v1 and a few others threw `RuntimeError: "rsqrt_cpu" not implemented for 'Half'`
- alexnet and ssd_vgg_16 threw `RuntimeError: expected scalar type Float but found Half`
- roberta_base_squad threw `RuntimeError: "LayerNormKernelImpl" not implemented for 'Half'`

Looks like `half()` only works on GPUs.

**Comment by kkontny** [comment]
Yes, it is expected. Since x86 doesn't support fp16 natively, nobody cared about this on CPU. However Altra supports it natively, so when eager mode will be implemented I expect it to work.
